### ISSUE TITLE
Missing application icon

### ISSUE THREAD

**Issue by RadekCap** [thinkingrock-gtd/tr-pc#28]
When you run an application compiled from the source it has a Netbeans icon instead of a TR icon during switching applications.

**Comment by ursjoss** [comment]
This is not the case on my system. But I have a suspicion. @RadekCap Please check the following:

- In NetBeans - with the project open - open the "Branding" dialog (Right-click on the the "ThinkingRock" node in the "Projects" group -> "Branding..." -> "Basic" Tab)
- You can see the place-holder for icons in 6 sizes (16x16 - 1024x1024).Clicking on each will show you the icons chosen for that size.


![Image](https://user-images.githubusercontent.com/357238/213028361-04db7cc9-b109-4842-bbb2-3e6890576f16.png)


As you can see, we have actual ThinkingRock icons in the sizes 16x16 to 48x48.However, for 256x256 and larger icons, we don't have an icon specified, falling back to the default Netbeans icon:


![Image](https://user-images.githubusercontent.com/357238/213028545-9cbd4871-57d3-4304-b3b7-8c75fa157752.png)

Can you experiment by selecting a different random icon for each of those missing sizes and check if that changes the situation on your system? I suspect that the screen resolution and/or pixel depth of your system falls back to one of the larger icons, which is not defined.

In case you can confirm that, we should try to get the icon in the missing sizes from Claire and Jeremy (or alternatively as a vector graphic, so we can convert it to the pngs we need ourselves.

**Comment by RadekCap** [comment]
I've tried to create a larger icon without any effect on the application icon (on MacOS).

I can observe this behavior both on Linux and MacOS.

I'm going to experiment with this later.

**Comment by RadekCap** [comment]
Ehm, I tried it again on Linux and a replacement 256x256 icon will solve it.

Maybe I haven't used a proper format on macOS, I'll explore it more.

**Comment by ursjoss** [comment]
I created a request for larger icons in ticket #67.

**Comment by RadekCap** [comment]
This solution doesn't work on MacOS. So it'll need another exploration and fix.

**Comment by RadekCap** [comment]
Following this guide should fix the issue. I'll look into it later when there will be more time.

https://netbeans.apache.org/wiki/DevFaqExecutableIcon.html

**Comment by ursjoss** [comment]
> Maybe it's irrelevant, but I can confirm that I don't see the TR icon; however, I don't see either the Netbeans icon, but something like this one (it's a bit different, but you get the idea):
> 
> ![image](https://user-images.githubusercontent.com/91336981/214047733-62e48b7a-462d-4b39-b48d-a4c633b62aba.png)

I think the installer that I provided was built before #68 was merged to main. I will update it and let you know to get the newer one when there...

**Comment by RadekCap** [comment]
@degrown is working on macOS where this bug is still valid, and I need to investigate it more. It's fixed on Linux.

The icon is the OpenJDK icon.

**Comment by thegamer87** [comment]
Hi,
trying to create the bundle for mac os x (through "ant build-mac") I noticed that the icon copied in the bundle is: netbeans-plat/17/harness/etc/applicationIcon.icns, trying to change this icon and doing a new build the bundle comes out with the updated icon, some simple google search point me to this netbeans wiki abount "DevFaqExecutableIcon":
https://netbeans.apache.org/wiki/DevFaqExecutableIcon.html

I hope this can help you to solve the problem.

**Comment by ursjoss** [comment]
Hi @thegamer87 

Thanks for your interest in TR, your report and investigation. The link looks relevant - even though on first glance, it looks like there is no out-of the box solution available from Netbeans. We may be able to implement a workaround though.

**Comment by thegamer87** [comment]
I've tried following the instructions in the link under "Mac Icons" and it seems to work, specifically I have

1. added correct trgtd icon ".icns" file (converted from one of .gif file used by linux application) in a subfolder of branding (obviously the file can be placed anywhere so it is then referenced in the build.xml, see next step)
`
➜  thinkingrock_git git:(main) ls -l branding/MacOS
total 136
-rw-r--r--@ 1 verrocchiom  staff  66774 Mar 20 08:59 trgtd.icns
`

2. added to build.xml the descriptor from the documentation, this override build-mac target to replace the default .icns with my .icns (see attached patch file of build.xml)

In this way the bundle come out with my .icns instead of default .icns.

[tr_build.xml.patch](https://github.com/thinkingrock-gtd/tr-pc/files/11015564/tr_build.xml.patch)

**Comment by ursjoss** [comment]
@thegamer87 Yes, I was thinking of something like that. Would you like to create a PR with your already established solution?

**Comment by thegamer87** [comment]
I've opened a PR #127 with more simple but out-of-the-box working solution (that not require any override in build manifest): digging into the build.mac target of suite.xml I've found that MacOS application icon is already read from the "app.icon.icns" property so I've simpy added this to _nbproject/project.properties_ pointing to .icns file that i've added in branding/MacOS folder

**Comment by ursjoss** [comment]
See also https://github.com/tibirna/qgit/pull/142 for an issue (and a solution) regarding missing icons on Mac.
